lib/mongoose-store.js

```diff
@@ -15,6 +15,33 @@ function convertToJSON(mongooseDocument) {
   return mongooseDocument ? mongooseDocument.toJSON() : undefined;
 }
 
+function errorHandler(data, operation) {
+  return function(err) {
+    if(err) {
+      var error = {
+        operation: operation,
+        name: err.name,
+        message: err.message,
+        document: data
+      };
+
+      if(err.errors) {
+        error.cause = {};
+        for(var e in err.errors) {
+          error.cause[e] = {
+            name: err.errors[e].name,
+            msg: err.errors[e].message
+          }
+        }
+      } else {
+        error.cause = err.message;
+      }
+
+      console.error("[ERROR] " + error.operation + ": [" + error.name + "] " + error.message + "\n", error);
```

Comment:
console can be passed as optional parameter here. When using console I think we should avoid formating log message as this may introduce some log inconsistency when using external logger format.